sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -815,13 +821,26 @@ def get_cmis_media_lanes_mask(self, api, appl, lport, subport):
 
         return media_lanes_mask
 
-    def is_appl_reconfigure_required(self, api, app_new):
+    def is_decommission_required(self, lport, api):
```

Comment:
@AnoopKamath given the fact that the default application is not matching with configured speed, is it safe to say that we need to decommission all 8 lanes for both:-

1. No breakout
2. Subport with same application
3. Subport with mixed application

My understanding its safe decommission all 8 lanes (for all 3 above cases) because I can't think of a mixed application as the default. If not, can you give me an example?